src/server/scripts/Zandalar/AtalDazar/boss_priestess_alun_za.cpp

```diff
@@ -0,0 +1,684 @@
+    SPELL_TAINTED_BLOOD_SELECTOR = 255592,
+    SPELL_TAINTED_BLOOD_MISSILE  = 255565,
+    SPELL_TAINTED_BLOOD_DAMAGE   = 255558,
+    SPELL_TAINTED_BLOOD_AT       = 255559,
+    SPELL_SPIRIT_OF_GOLD         = 259205,
+    SPELL_CORRUPTED_GOLD         = 258703,
+    SPELL_CORRUPTED_GOLD_DAMAGE  = 258709,
+
+    SPELL_SPAWN                  = 259018,
+    SPELL_TIKI_MASK              = 277065,
+    SPELL_CORRUPT                = 259032,
+    SPELL_FATALLY_CORRUPTED      = 259123
+};
+
+enum PriestessAlunzaEvents
+{
+    EVENT_GILDED_CLAWS = 1,
+    EVENT_TAINTED_BLOOD,
+    EVENT_MOLTEN_GOLD,
+    EVENT_CHECK_ENERGY,
+    EVENT_CORRUPTED_GOLD,
+    EVENT_SPIRIT_OF_GOLD,
+
+    EVENT_CHECK_TAINTED_BLOOD
+};
+
+enum PriestessAlunzaTexts
+{
+    SAY_AGGRO                = 0,
+    SAY_GILDED_CLAWS_WARNING = 1,
+    SAY_GILDED_CLAWS         = 2,
+    SAY_TRANSFUSION_WARNING  = 3,
+    SAY_TRANSFUSION          = 4,
+    SAY_SLAY                 = 5,
+    SAY_WIPE                 = 6,
+    SAY_DIED                 = 7
+};
+
+enum PriestessAlunzaMisc
+{
+    NPC_CORRUPTED_GOLD        = 130738,
+    NPC_SPIRIT_OF_GOLD        = 131009,
+
+    POINT_DESPAWN             = 0,
+    POINT_POOL                = 1,
+
+    AREATRIGGER_TAINTED_BLOOD = 16553
+};
+
+constexpr Position RitualPosition[5] =
+{
+    { -1124.4688f, 2294.139f,  747.2801f  },
+    { -1126.8889f, 2307.7466f, 743.11414f },
+    { -1124.342f,  2292.5173f, 747.76434f },
+    { -1131.6841f, 2304.0347f, 743.94354f },
+    { -1131.2916f, 2282.3438f, 743.99896f }
+};
+
+constexpr Position TaintedBloodPosition[11] =
+{
+    { -1084.0f, 2276.5f, 740.784f   },
+    { -1084.0f, 2292.5f, 740.784f   },
+    { -1096.0f, 2316.5f, 740.78394f },
+    { -1088.0f, 2268.5f, 740.784f   },
+    { -1100.0f, 2276.5f, 740.78406f },
+    { -1096.0f, 2268.5f, 740.784f   },
+    { -1080.0f, 2316.5f, 740.78394f },
+    { -1088.0f, 2284.5f, 740.784f   },
+    { -1088.0f, 2316.5f, 740.604f   },
+    { -1080.0f, 2300.5f, 740.603f   },
+    { -1100.0f, 2308.5f, 740.604f   }
+};
+
+constexpr Position CorruptedGoldSpawnPosition[2] =
+{
+    { -1110.4965f, 2259.4306f, 741.8436f, 0.261799f },
+    { -1110.5451f, 2325.2829f, 741.8436f, 0.785398f }
+};
+
+constexpr Position SpiritOfGoldSpawnPosition = { -1119.8f, 2292.46f, 745.114f };
+
+// 122967 - Priestess Alun'za
+struct boss_priestess_alun_za : public BossAI
+{
+    boss_priestess_alun_za(Creature* creature) : BossAI(creature, DATA_PRIESTESS_ALUNZA) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_RITUAL);
+        me->SetPowerType(POWER_ENERGY);
+        me->SetPower(POWER_ENERGY, 0);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        summons.DespawnAll();
+
+        Talk(SAY_WIPE);
+
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim->IsPlayer())
```

Comment:
I suggest using some type of time_t to prevent spamming kill texts. Some type of 3-5s of difference should be ok.